From 3493d3514d16b1cc73c62f968113fbd0a6f7071c Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Tue, 19 Aug 2014 19:30:42 -0400
Subject: [scsi] scsi_transport_iscsi: Export ISCSI_PARAM_LOCAL_IPADDR attr for iscsi_connection

Message-id: <1408476658-5056-5-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88340
O-Subject: [PATCH RHEL 7.1 e-stor 04/20] scsi_transport_iscsi: Export ISCSI_PARAM_LOCAL_IPADDR attr for iscsi_connection
Bugzilla: 1089349
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Adheer Chandravanshi <adheer.chandravanshi@qlogic.com>

scsi commit id 5af62f240b4bf6a79ffe71d570b7b1e2765ea0d0

This attribute specifies the local IP address used to establish connection.

Signed-off-by: Adheer Chandravanshi <adheer.chandravanshi@qlogic.com>
Signed-off-by: Vikas Chaudhary <vikas.chaudhary@qlogic.com>
Reviewed-by: Mike Christie <michaelc@cs.wisc.edu>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/scsi_transport_iscsi.c b/drivers/scsi/scsi_transport_iscsi.c
index 4cf918a..fd8ffe6 100644
--- a/drivers/scsi/scsi_transport_iscsi.c
+++ b/drivers/scsi/scsi_transport_iscsi.c
@@ -3772,6 +3772,7 @@ iscsi_conn_attr(ipv6_flow_label, ISCSI_PARAM_IPV6_FLOW_LABEL);
 iscsi_conn_attr(is_fw_assigned_ipv6, ISCSI_PARAM_IS_FW_ASSIGNED_IPV6);
 iscsi_conn_attr(tcp_xmit_wsf, ISCSI_PARAM_TCP_XMIT_WSF);
 iscsi_conn_attr(tcp_recv_wsf, ISCSI_PARAM_TCP_RECV_WSF);
+iscsi_conn_attr(local_ipaddr, ISCSI_PARAM_LOCAL_IPADDR);
 
 
 #define iscsi_conn_ep_attr_show(param)     \
@@ -3841,6 +3842,7 @@ static struct attribute *iscsi_conn_attrs[] = {
  &dev_attr_conn_is_fw_assigned_ipv6.attr,
  &dev_attr_conn_tcp_xmit_wsf.attr,
  &dev_attr_conn_tcp_recv_wsf.attr,
+ &dev_attr_conn_local_ipaddr.attr,
  NULL,
 };
 
@@ -3910,6 +3912,8 @@ static umode_t iscsi_conn_attr_is_visible(struct kobject *kobj,
   param = ISCSI_PARAM_TCP_XMIT_WSF;
  else if (attr == &dev_attr_conn_tcp_recv_wsf.attr)
   param = ISCSI_PARAM_TCP_RECV_WSF;
+ else if (attr == &dev_attr_conn_local_ipaddr.attr)
+  param = ISCSI_PARAM_LOCAL_IPADDR;
  else {
   WARN_ONCE(1, "Invalid conn attr");
   return 0;
diff --git a/include/scsi/iscsi_if.h b/include/scsi/iscsi_if.h
index 3851a73..fd0421c 100644
--- a/include/scsi/iscsi_if.h
+++ b/include/scsi/iscsi_if.h
@@ -618,6 +618,7 @@ enum iscsi_param {
 
  ISCSI_PARAM_DISCOVERY_PARENT_IDX,
  ISCSI_PARAM_DISCOVERY_PARENT_TYPE,
+ ISCSI_PARAM_LOCAL_IPADDR,
  /* must always be last */
  ISCSI_PARAM_MAX,
 };
-- 
1.7.1